### ISSUE TITLE
PLE2502: False positive for escaped unicode names ``'\N{...}'``

### ISSUE THREAD

**Issue by AA-Turner** [charliermarsh/ruff#2679]
The new bidirectional unicode detector (#2589) doesn't seem to recognise ``\N{...}`` escapes (https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals), see below:--

```powershell
(sphinx) PS I:\Development\sphinx> type bug.py
print('\N{RIGHT-TO-LEFT MARK}')
(sphinx) PS I:\Development\sphinx> pylint --version
pylint 2.16.1
astroid 2.14.1
Python 3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:12:32) [MSC v.1929 64 bit (AMD64)]
(sphinx) PS I:\Development\sphinx> ruff -V                                   
ruff 0.0.244
(sphinx) PS I:\Development\sphinx> pylint --disable=all --enable E2502 bug.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

(sphinx) PS I:\Development\sphinx> ruff --isolated --select PLE2502 bug.py   
bug.py:1:7: PLE2502 Avoid using bidirectional unicode
Found 1 error.
(sphinx) PS I:\Development\sphinx>
```

Ruff version 0.0.244

Pylint does not raise the error (see part one of the reproducer).

A

**Comment by AA-Turner** [comment]
Copying @colin99d as the author of the new check.

A

**Comment by charliermarsh** [comment]
Thanks for filing.
